tools/build/sprite/player_sprites.py

```diff
@@ -0,0 +1,466 @@
+                    sheet_rtes_back.append(
+                        RasterTableEntry(
+                            png_info.offset, png_info.width * png_info.height // 2
+                        )
+                    )
+                else:
+                    sheet_rtes_back.append(RasterTableEntry(SPECIAL_RASTER, 0x10))
+
+            png_info = RASTER_CACHE[raster_xml.attrib["src"][:-4]]
+            sheet_rtes.append(
+                RasterTableEntry(png_info.offset, png_info.width * png_info.height // 2)
+            )
+
+        raster_info_offsets.append(len(rtes))
+        num_sheets += 1
+        rtes.extend(sheet_rtes)
+        if has_back:
+            raster_info_offsets.append(len(rtes))
+            num_sheets += 1
+            rtes.extend(sheet_rtes_back)
+    raster_info_offsets.append(len(rtes))  # Final 'offset' (size of list)
+
+    separators_offset = 0x10
+    infos_offset = separators_offset + (num_sheets + 1) * 4
+    rasters_offset = infos_offset + (len(rtes) + 1) * 4
+
+    # Align raster_offset_start to 0x10 offset
+    rasters_offset = (rasters_offset + 0xF) & ~0xF
+
+    for rte in rtes:
+        if rte.offset == SPECIAL_RASTER:
+            built_raster_info += struct.pack(">I", 0x0011F880)
+            continue
+
+        packed_info = (rte.size >> 4) << 20
+        packed_info |= (rte.offset + rasters_offset) & 0xFFFFF
+        built_raster_info += struct.pack(">I", packed_info)
+
+    # This is the missing raster from before
+    built_raster_info += struct.pack(">I", 0x06C9CD50)
+
+    header = struct.pack(">IIII", separators_offset, infos_offset, rasters_offset, 0)
+
+    info_list_bytes = b""
+    for offset in raster_info_offsets:
+        info_list_bytes += struct.pack(">I", offset)
+
+    built_raster_info = header + info_list_bytes + built_raster_info
+
+    # Align build_raster_info to 0x10 offset
+    built_raster_info += b"\0" * ((0x10 - len(built_raster_info)) & 0xF)
+
+    raster_bytes = b""
+    for raster_name in raster_order[:-1]:  # Skip last raster
+        png_info = RASTER_CACHE[raster_name]
+        raster_bytes += png_info.data
+
+    built_raster_info += raster_bytes
+
+    built_raster_info_offset = len(built_raster_info) + 0x10
+    compressed_sprite_bytes_offset = built_raster_info_offset + len(
+        compressed_sprite_bytes
+    )
+
+    major_file_divisons = struct.pack(
+        ">IIII",
+        0x10,
+        built_raster_info_offset,
+        compressed_sprite_bytes_offset,
+        0x23F1F8,  # TODO hard-coded
+    )
+
+    final_data = (
+        build_info_bytes
+        + major_file_divisons
+        + built_raster_info
+        + compressed_sprite_bytes
+    )
+
+    with open(out_binary, "wb") as f:
+        f.write(final_data)
+
+
+if __name__ == "__main__":
+    if len(sys.argv) != 3:
+        print("usage: player_sprites.py [OUTBIN] [DIR]")
+        exit(1)
+
+    _, outfile, sprite_dir = sys.argv
+    build(Path(sprite_dir), Path(outfile))
```

Comment:
Instead of passing a single directory, can we pass all source files or a set of directories that will be layered? See #1053

Reply:
How do we influence the order properly? I don't really understand how this would work.

Reply:
Specifically it would look a bit like
```
player_sprites.py [OUTBIN] assets/us/sprite/player/SpriteSheet.xml assets/us/sprite/player/MyRaster.png assets/mymod/sprite/player/NewRaster.png
```
where are file stems are unique but if a file exists higher up the asset stack we prefer it. there's a function in configure.py for this